lib/node_modules/@stdlib/math/base/complex/cis/benchmark/python/benchmark.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+"""Benchmark cis."""
+
+from __future__ import print_function
+import timeit
+
+NAME = "cis"
+REPEATS = 3
+ITERATIONS = 1000000
+
+
+def print_version():
+    """Print the TAP version."""
+    print("TAP version 13")
+
+
+def print_summary(total, passing):
+    """Print the benchmark summary.
+
+    # Arguments
+
+    * `total`: total number of tests
+    * `passing`: number of passing tests
+
+    """
+    print("#")
+    print("1.." + str(total))  # TAP plan
+    print("# total " + str(total))
+    print("# pass  " + str(passing))
+    print("#")
+    print("# ok")
+
+
+def print_results(elapsed):
+    """Print benchmark results.
+
+    # Arguments
+
+    * `elapsed`: elapsed time (in seconds)
+
+    # Examples
+
+    ``` python
+    python> print_results(0.131009101868)
+    ```
+    """
+    rate = ITERATIONS / elapsed
+
+    print("  ---")
+    print("  iterations: " + str(ITERATIONS))
+    print("  elapsed: " + str(elapsed))
+    print("  rate: " + str(rate))
+    print("  ...")
+
+
+def benchmark():
+    """Run the benchmark and print benchmark results."""
+    setup = "from random import random; from numpy import exp;"
+    stmt = "y = exp(1.0j * ( ( (random()*100.0) - 50.0 ) + 1.0j * ( ( (random()*100.0) - 50.0 ) ) ) )"
```

Comment:
Should we update this benchmark similar to how "cis" is computed in the C benchmark?

Reply:
I've modified this as follows:

```python
re = ( random() * 100.0 ) - 50.0;
im = ( random() * 100.0 ) - 50.0;
y = ( cos( re ) + 1.0j * sin( re ) ) / exp( im );
```

Which, fwiw is probably *very rarely* the way you'd actually perform, say, a million computations with numpy. You'd probably vectorize it like this instead:

```python
from numpy import sin, cos, exp, random;
re = ( random.rand(1000000) * 100.0 ) - 50.0;
im = ( random.rand(1000000) * 100.0 ) - 50.0;
y = ( cos( re ) + 1.0j * sin( re ) ) / exp( im );
```

Not a big deal, just a small note on translation vs. transliteration for idiomatic benchmarks. I have no strong feelings here though, as long as the subtlety is not lost.

Reply:
Yeah, here, the benchmark is **not** for a million values; the benchmark is for a single function call to compute a single value. Were we benchmarking `cis` in JavaScript where the function was provided an `ndarray`, yes, you are correct, this would not be an apples-to-apples comparison. However, similarly, were we to "vectorize" the Python benchmark, we would be giving Python an unfair advantage and, thus, not be providing an apples-to-apples comparison.

When we provide a generic API for `cis`, then, yes, it will be appropriate to provide Python (and R and Julia) benchmarks which "vectorize". But, here, no.

Reply:
I couldn't help but cringe at writing code to run it separately, but that makes perfect sense. 👍 

✅ on this then, I believe.

Reply:
Understood. Just have to always keep in mind what you are benchmarking. ;)

Reply:
For the record, I updated the benchmark to not use `numpy` and, instead, to just use the standard math library functions. (apples-to-apples)

Reply:
Ah, no problem. Also makes sense. 👍